src/main/kotlin/org/rust/cargo/project/model/impl/CargoProjectImpl.kt

```diff
@@ -341,7 +341,9 @@ data class CargoProjectImpl(
     }
 
     override val presentableName: String
-        get() = workingDirectory.fileName.toString()
+        get() = projectService.allPackages.firstOrNull {
```

Comment:
Am I understand right that package name should be only used in case of a single-package project? If so, `singleOrNull` should be used instead, or even `.singleOrNull()?.takeIf { ... }`

Reply:
As I understand, here I need to find a package with the same directory as the project one.

There is a mistake, instead of just trying first I think I actually should filter one from all by working directory.

Reply:
`.find` and `.firstOrNull` are equivalent and IMO `.firstOrNull` is preferable because of very descriptive name. Look at the definition in Kotlin stdlib:

```Kotlin
public inline fun <T> Iterable<T>.find(predicate: (T) -> Boolean): T? {
    return firstOrNull(predicate)
}
```

But I'm not sure that we should use the name of **some** package as a cargo project name in case of **multiple** packages

Reply:
I think I can't imagine how multiple packages can have the same working directory.

If we have a Cargo.toml with a name inside it, then we can find it in the packages list by working directory and show its name instead of its directory. Otherwise, when there is no packages **with the same working directory** (so project directory doesn't contain any root package), we fall back to working directory name.

Reply:
I think current approach in general